cmd/internal/backup/backup.go

```diff
@@ -105,6 +109,15 @@ func (b *Backuper) CreateBackup(ctx context.Context) error {
 
 	b.log.Info("compressed backup")
 
+	if b.encrypter != nil {
+		filename, err = b.encrypter.Encrypt(filename)
+		if err != nil {
+			b.metrics.CountError("encrypt")
+			return fmt.Errorf("error encrypting backup: %w", err)
+		}
+		b.log.Info("encrypted backup")
+	}
+
 	err = b.bp.UploadBackup(ctx, filename)
```

Comment:
I wonder what happens if someone change from plain -> encrypted backups or from encrypted -> plain. Do we need to prevent certain scenarios ? should we warn the user ?

Reply:
I talked with @Gerrit91 about this:

plain -> encrypted: no problem since encryption is an opt-in feature
encrypted -> plain: error will be thrown: https://github.com/metal-stack/backup-restore-sidecar/blob/7a6cd7922ce43f9effed0d4d7b37f594e03668de/cmd/internal/encryption/encryption.go#L191

Solution for this could be referencing this in the Readme -> and tell that manuell decryption is an open feature

Reply:
But even when switching from plain -> encrypted, old backups are not encrypted, are they unpacked without trying to decrypt them ? If this is the case i agree

For the other case referencing to the Readme is not sufficient because no one reads them.

Reply:
One Idea would be to check if the last backup is encrypted and throw an error if plaintext is selected ?

Reply:
I've now implemented both cases:

1. plain -> encrypted:  encryption works without problem / downloading & restoring now don't decrypt if file is not encrypted
2. encrypted -> plain: error on initialization checking if latest backup is encrypted, but no encryption is configured